Assets/Thirdweb/Examples/Scripts/Prefab_Events.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections.Generic;
+using UnityEngine;
+using Thirdweb;
+
+// Your Event data structure
+[System.Serializable]
+public struct TransferEvent
+{
+    public string from;
+    public string to;
+    public string tokenID;
+
+    public override string ToString()
+    {
+        return
+            $"TransferEvent:"
+            + $"\n>prevURI: {from}"
+            + $"\n>newURI: {to}"
+            + $"\n>tokenID: {tokenID}";
+    }
+}
+
+
+public class Prefab_Events : MonoBehaviour
+{
+
+    // Get all events filtered by name (and optionally add more filters)
+
+    public async void GetEventsTest()
+    {
+        Contract contract = new Contract("goerli", "0x2e01763fA0e15e07294D74B63cE4b526B321E389");
+
+        List<ContractEvent<TransferEvent>> allEvents = await contract.events.Get<TransferEvent>("Transfer");
```

Comment:
wait so passing `tokenId = 0` doesnt work here? (with the corrected event property name)

Reply:
it does, you can still pass it as a string, problem is with a type that contains fromBlock, toBlock and order + the filter dictionary, only using one of these sets the rest to default, which if fromBlock and toBlock is not set, they're 0

Sticking to a string solves this for now while I work on a different API with those filters as separate parameters to the direct Get and GetAll calls